File: 01_strings_and_arrays/02_compress/compress.js
const compress = (s) => {
    let str = '';
    let i = 0;
    let j = 0;
    
    while(j < s.length) {
      if(s[j] === s[i]) {
        j++;
      }
      if(s[j] != s[i] || j == s.length){
        let num = j - i
        if(num == 1) {
          str += s[i];
          i = j;
        } else {
          str += num.toString() + s[i];
          i = j; 
        }
      }
    }
        
    return str;
  };

File: 01_strings_and_arrays/02_compress/compress.py
def compress(s):
  s += '!'
  res = ''
  i = 0
  j = 0
  
  while j < len(s):
    if s[j] == s[i]:
      j += 1
    else:
      num = j - i
      if num == 1:
        res += s[i]
        i = j
      else:
        res += str(num) + s[i]
        i = j
        
  return res